lib.php

```diff
@@ -0,0 +1,61 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Libs.
+ * @copyright  Catalyst IT 2022
+ * @package    local_login
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * Triggers after site config is loaded.
+ *
+ */
+function local_login_after_config() {
+    global $CFG, $SESSION, $_SERVER;
+    $errorcode = 0;
+    $noredirect  = optional_param('noredirect', 0, PARAM_BOOL); // don't redirect
+
+    if (!isloggedin() && empty($noredirect)) {
```

Comment:
shouldn't that be !empty ? -

Reply:
I'm also inclined to suggest we add something into that if check that says ... "and we are on the login/index.php page"

so - the check goes - if we're not logged in, and if we're on the core login/index.php page, and if noredirect = 1, unset the alternateloginurl so that the core login page loads even if alternateloginurl is configured.

Reply:
The thing is that the login page calls itself, it reloads in the process of logging which is where the noredirect variable gets lost. Which is why the check does not work as intended. 

I tried with your suggestions but it did not work. The login does not happen, it keeps directing to the local/login page.